web/src/compose_validate.ts

```diff
@@ -54,6 +54,20 @@ function set_message_too_long(status: boolean): void {
     update_send_button_status();
 }
 
+function set_message_too_long_for_edit(status: boolean, $container: JQuery): void {
+    message_too_long = status;
+    const $message_edit_save_container = $container.find(".message_edit_save_container");
+    if (message_too_long) {
+        $container.find(".message_edit_save").prop("disabled", true);
```

Comment:
I'm having a bit of trouble figuring out if this logic always works.

A cleaner, slightly different version is:

```
const save_is_disabled = message_too_long || $message_edit_save_container.hasClass("message-edit-time-limit-expired");

$container.find(".message_edit_save").prop("disabled", save_is_disabled);
$message_edit_save_container.toggleClass("disabled-message-edit-save", save_is_disabled);
```

If you think that would also do the right thing, let's switch to that.

Reply:
Yup, that was a nice refactor! Thanks.